### PR TITLE
Fixes for VTK layout

### PR DESCRIPTION
**Pull request by philippjfr**
Fixes https://github.com/holoviz/panel/issues/987

### CODE REVIEW

File: panel/models/vtk/vtk.ts [hunk 1]
```diff
@@ -79,48 +79,43 @@ export class VTKPlotView extends PanelHTMLBoxView {
...
-      })
-      this._plot()
-      this._rendererEl.getRenderer().getActiveCamera().onModified(() => this._get_camera_state())
-      this._remove_default_key_binding()
-      this.model.renderer_el = this._rendererEl
-    }
-    super.after_layout()
+  render(): void {
+    super.render()
+    const container = div()
```
Comment (xavArtley) [inline panel/models/vtk/vtk.ts]: Not sure If the div is really necessay or if we can replace it by this.el and the vtk root container by this.root.el

Comment (philippjfr) [reply panel/models/vtk/vtk.ts]: I think in certain grid plots there might not be any actual guarantee `this.root.el` exists.

Comment (philippjfr) [reply panel/models/vtk/vtk.ts]: Generally other components also seem to add a subcontainer to `this.el` so I'm inclined to leave it.

Comment (xavArtley) [reply panel/models/vtk/vtk.ts]: ok

File: panel/models/vtk/vtk.ts [hunk 2]
```diff
@@ -79,48 +79,42 @@ export class VTKPlotView extends PanelHTMLBoxView {
...
+      const dims = container.getBoundingClientRect()
+      const width = Math.floor(dims.width * window.devicePixelRatio)
+      const height = Math.floor(dims.height * window.devicePixelRatio)
+      axes_canvas.setAttribute('width', width.toFixed())
+      axes_canvas.setAttribute('height', height.toFixed())
+    })
+    this._plot()
+    this._rendererEl.getRenderer().getActiveCamera().onModified(() => this._get_camera_state())
+    this._remove_default_key_binding()
+    this.model.renderer_el = this._rendererEl
```
Comment (philippjfr) [inline panel/models/vtk/vtk.ts]: @xavArtley While I'm looking at this code, I'm assuming this is added to make JS linking possible in some form?

Comment (philippjfr) [reply panel/models/vtk/vtk.ts]: It's usually bad form to attach anything related to the view to a model but if it's about linking I get why you're doing this.

Comment (xavArtley) [reply panel/models/vtk/vtk.ts]: Indeed

Comment (philippjfr) [reply panel/models/vtk/vtk.ts]: Thanks, glad I have at least a rudimentary understanding of the VTK models having done this.

**Pull request merged**